notification-portlet-webapp/src/main/webapp/scripts/modal-notice.js

```diff
@@ -0,0 +1,151 @@
+        }
+      });
+    }
+
+    // Then fetch the notifications with a ResourceURL
+    function fetchNotices($, settings, callback) {
+      $.ajax({
+        url      : settings.getNotificationsUrl,
+        type     : 'POST',
+        dataType : 'json',
+        success: function (data) {
+          feed = data.feed;
+          callback(feed);
+        },
+        error: function () {
+          container.html(" ").text("AJAX failed. ~ THE END ~");
+        }
+      });
+
+    }
+
+    upmodal_notice.show = function ($, container, options) {
+
+      var settings = $.extend({}, defaults, options);
+
+      var drawActions = function(actionsContainer, alert) {
+
+          var availableActions = alert.availableActions;
+          var actionTemplate = actionsContainer.find(settings.selectors.actionTemplate);
+
+          for (var i=0; i < availableActions.length; i++) {
+              var action = availableActions[i];
+
+              var actionUrl = settings.invokeActionUrlTemplate
+                      .replace('NOTIFICATIONID', alert.id)
+                      .replace('ACTIONID', action.id);
+
+              var actionElement = actionTemplate.clone();
+              actionElement.removeClass('action-template');
+              actionElement.toggleClass('hidden');
+              actionElement.addClass('action');
+              actionElement.find('a').attr('href', actionUrl).html("ACCEPT");
+              actionElement.appendTo(actionsContainer);
+          }
+
+          actionsContainer.toggleClass('hidden');
+
+      }
+
+      var drawNotices = function(feed) {
+
+        // Do we have any notices to show?
+        if (feed && feed.length != 0) {
+
+          // Iterate the notices
+          var len = feed.length;
+          for (var i=0; i < len; i++) {
+            var alert = feed[i];
+
+            var element = container.find(settings.selectors.content);
+
+            // Insert context
+            element.find(settings.selectors.title).html(alert.title);
+            if (alert.body) {
+              element.find(settings.selectors.body).html(alert.body);
+            }
+
+            if (alert.url) {
+              var linkText = alert.linkText || alert.url;
+              element.find(settings.selectors.link).attr('href', alert.url).html(linkText);
+            }
+
+            // Are actions available?
+            if (alert.availableActions && alert.availableActions.length != 0) {
+                var actionsContainer = element.find(settings.selectors.actions);
+                if (actionsContainer) {
+                    drawActions(actionsContainer, alert);
+                }
+            }
+
+          }
+
+          // Invoke the specified callback function, if any
+          settings.readyCallback();
+        }
+
+      }
+
+      // Invoke notifications
+      initNotices($, settings, drawNotices);
+
+    }
+
+    upmodal_notice.pullFeed = function ($, options, callback) {
```

Comment:
:question: why is jQuery being passed around as a function param?

Reply:
Not sure how it's being used here, but we normally do something like...

```
(function($) {
    // do stuff with access to $
})(up.jQuery);
```

Reply:
I've seen an IIFE that defines `$` (like in your example) in several places around uPortal.
Passing `$` as a param to every function is new to me.

Reply:
Yeah it's probably better -- all things considered -- not to pass it to each function.

Reply:
It is being passed in this manner because the jQuery version is determined in configuration (whether to use uPortal's version or Notification's own).

A big time sink was getting the right version of jQuery with Modal setup. 

Since this is the approach of the other JS files, I will continue with this. I suspect a front-end rewrite will clear much of this up.